packages/remix-dev/devServer_unstable/index.ts

```diff
@@ -267,6 +268,12 @@ export let serve = async (
 
   server.listen(options.port);
 
+  exitHook(() => {
+    state.appServer?.pid && kill(state.appServer.pid);
+    websocket.close();
+    server.close();
+  });
+
   return new Promise(() => {}).finally(async () => {
```

Comment:
What does this `Promise` do? As far as I've tested it, it never resolves or rejects.